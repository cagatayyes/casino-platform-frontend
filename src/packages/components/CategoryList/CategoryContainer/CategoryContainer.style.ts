import tw from 'tailwind-styled-components';

export const CategorySection = tw.div`
  flex
  flex-col
  justify-center
  bg-[#0D1335]
  mt-[28px]
  first-of-type:mt-[0]
  relative
`;

export const GameCardSection = tw.div`
  flex
  flex-row
  mt-[16px]
`;
