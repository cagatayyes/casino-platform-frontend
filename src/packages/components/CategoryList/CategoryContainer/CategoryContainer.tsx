import { GameCard, GameCardSlider, CategoryTopContainer } from 'casino/components/';
import { CategorySection, GameCardSection } from './CategoryContainer.style';
import { IGame } from 'casino/types';

interface ICategoryContainerProps {
	games: IGame[];
	sliderNo: number;
	title: string;
	viewAllLink: string;
	iconName: string;
	showgameinfomodal: (gameId: number, gameLink: string) => void;
	handlegameload: (gameLink: string, gameName: string, gamelaunchpoint: string, categoryName?: string) => void;
}

export const CategoryContainer = (props: ICategoryContainerProps): JSX.Element => {
	const {
		games,
		sliderNo,
		title,
		viewAllLink,
		iconName,
		showgameinfomodal,
		handlegameload
	} = props;

	return (
		<CategorySection data-testid='category-section'>
			<CategoryTopContainer title={title}
				viewAllLink={viewAllLink}
				iconName={iconName} />

			<GameCardSection>
				<GameCardSlider sliderNo={sliderNo}>
					{
						games && games.length > 0 &&
						games.map(game => (
							<GameCard key={game.id}
								gamelaunchpoint='Lobby'
								showamount={false}
								gamename={game.name}
								gameimage={game.image_url}
								sash={game.sash}
								gameid={game.id}
								showgameinfomodal={showgameinfomodal}
								handlegameload={handlegameload}
							/>
						))
					}
				</GameCardSlider>
			</GameCardSection>
		</CategorySection>
	);
};

