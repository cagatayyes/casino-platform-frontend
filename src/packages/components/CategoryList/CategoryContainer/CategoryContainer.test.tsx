import React from 'react';
import { render, screen } from '@testing-library/react';
import { CategoryContainer } from './CategoryContainer';
import { BrowserRouter } from 'react-router-dom';

/* eslint-disable react/prop-types */
jest.mock('swiper/react', () => ({
	Swiper: ({ children }):JSX.Element => <div>{children}</div>,
	SwiperSlide: ({ children }):JSX.Element => <div>{children}</div>
}));

jest.mock('swiper/modules', () => ({
	Swiper: ({ children }):JSX.Element => <div>{children}</div>,
	SwiperSlide: ({ children }):JSX.Element => <div>{children}</div>
}));

describe('CategoryContainer Component Tests', () => {
	const mockGames = [
		{ id: 1, name: 'Game 1', image_url: 'url1', sash: 'New' },
		{ id: 2, name: 'Game 2', image_url: 'url2', sash: 'Popular' }
	];

	it('CategoryContainer renders', () => {
		const component = render(
			<BrowserRouter>
				<CategoryContainer games={mockGames}
					sliderNo={0}
					title=""
					viewAllLink=""
					iconName=""
					showGameInfoModal={() => { }}
					handleGameLoad={jest.fn()}
				/>
			</BrowserRouter>,
		);

		expect(screen.getByTestId('category-section')).toBeInTheDocument();
		expect(component).toMatchSnapshot();
	});
});
