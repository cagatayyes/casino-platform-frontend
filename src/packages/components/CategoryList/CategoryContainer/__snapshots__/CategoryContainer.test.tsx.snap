// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryContainer Component Tests CategoryContainer renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col justify-center bg-[#0D1335] mt-[28px] first-of-type:mt-[0] relative"
        data-testid="category-section"
      >
        <div
          class="flex flex-row items-center"
        >
          <span
            class="flex text-[16px] text-[#FFF] font-semibold truncate w-1/2"
          />
        </div>
        <div
          class="flex flex-row mt-[16px]"
        >
          <div
            class="flex overflow-hidden"
          >
            <div>
              <div>
                <div
                  class="flex flex-col w-[113px] h-[168px] bg-[#070C26] rounded-[8px] last-of-type:mr-[0]"
                  style="width: 113px;"
                >
                  <div
                    class="flex flex-col relative items-center h-[147px]"
                  >
                    <div
                      class="flex justify-center items-center px-[4px] w-fit h-[14px] rounded-[11px] top-[3px] outline outline-[0.5px] outline-[#00000066] drop-shadow-sash right-[3px] absolute"
                    >
                      <span
                        class="flex text-[12px] text-[#ffffff] mr-[1px] material-symbols-outlined"
                      />
                      <span
                        class="flex text-[#ffffff] text-[10px] font-semibold"
                      />
                    </div>
                    <img
                      class="flex rounded-[8px] h-full w-full"
                      src="url1"
                    />
                  </div>
                  <div
                    class="flex flex-row justify-between items-center h-[24px] px-[6px]"
                  >
                    <div
                      class="block font-bold text-[10px] text-[#4C5576] overflow-hidden whitespace-nowrap text-ellipsis w-[95px]"
                    >
                      Game 1
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="flex flex-col w-[113px] h-[168px] bg-[#070C26] rounded-[8px] last-of-type:mr-[0]"
                  style="width: 113px;"
                >
                  <div
                    class="flex flex-col relative items-center h-[147px]"
                  >
                    <div
                      class="flex justify-center items-center px-[4px] w-fit h-[14px] rounded-[11px] top-[3px] outline outline-[0.5px] outline-[#00000066] drop-shadow-sash right-[3px] absolute"
                    >
                      <span
                        class="flex text-[12px] text-[#ffffff] mr-[1px] material-symbols-outlined"
                      />
                      <span
                        class="flex text-[#ffffff] text-[10px] font-semibold"
                      />
                    </div>
                    <img
                      class="flex rounded-[8px] h-full w-full"
                      src="url2"
                    />
                  </div>
                  <div
                    class="flex flex-row justify-between items-center h-[24px] px-[6px]"
                  >
                    <div
                      class="block font-bold text-[10px] text-[#4C5576] overflow-hidden whitespace-nowrap text-ellipsis w-[95px]"
                    >
                      Game 2
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="navigationButton navigationButtonPrev navigationButtonPrev-0 material-symbols-outlined"
            >
              arrow_left_alt
            </span>
            <span
              class="navigationButton navigationButtonNext navigationButtonNext-0 material-symbols-outlined"
            >
              arrow_right_alt
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col justify-center bg-[#0D1335] mt-[28px] first-of-type:mt-[0] relative"
      data-testid="category-section"
    >
      <div
        class="flex flex-row items-center"
      >
        <span
          class="flex text-[16px] text-[#FFF] font-semibold truncate w-1/2"
        />
      </div>
      <div
        class="flex flex-row mt-[16px]"
      >
        <div
          class="flex overflow-hidden"
        >
          <div>
            <div>
              <div
                class="flex flex-col w-[113px] h-[168px] bg-[#070C26] rounded-[8px] last-of-type:mr-[0]"
                style="width: 113px;"
              >
                <div
                  class="flex flex-col relative items-center h-[147px]"
                >
                  <div
                    class="flex justify-center items-center px-[4px] w-fit h-[14px] rounded-[11px] top-[3px] outline outline-[0.5px] outline-[#00000066] drop-shadow-sash right-[3px] absolute"
                  >
                    <span
                      class="flex text-[12px] text-[#ffffff] mr-[1px] material-symbols-outlined"
                    />
                    <span
                      class="flex text-[#ffffff] text-[10px] font-semibold"
                    />
                  </div>
                  <img
                    class="flex rounded-[8px] h-full w-full"
                    src="url1"
                  />
                </div>
                <div
                  class="flex flex-row justify-between items-center h-[24px] px-[6px]"
                >
                  <div
                    class="block font-bold text-[10px] text-[#4C5576] overflow-hidden whitespace-nowrap text-ellipsis w-[95px]"
                  >
                    Game 1
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                class="flex flex-col w-[113px] h-[168px] bg-[#070C26] rounded-[8px] last-of-type:mr-[0]"
                style="width: 113px;"
              >
                <div
                  class="flex flex-col relative items-center h-[147px]"
                >
                  <div
                    class="flex justify-center items-center px-[4px] w-fit h-[14px] rounded-[11px] top-[3px] outline outline-[0.5px] outline-[#00000066] drop-shadow-sash right-[3px] absolute"
                  >
                    <span
                      class="flex text-[12px] text-[#ffffff] mr-[1px] material-symbols-outlined"
                    />
                    <span
                      class="flex text-[#ffffff] text-[10px] font-semibold"
                    />
                  </div>
                  <img
                    class="flex rounded-[8px] h-full w-full"
                    src="url2"
                  />
                </div>
                <div
                  class="flex flex-row justify-between items-center h-[24px] px-[6px]"
                >
                  <div
                    class="block font-bold text-[10px] text-[#4C5576] overflow-hidden whitespace-nowrap text-ellipsis w-[95px]"
                  >
                    Game 2
                  </div>
                </div>
              </div>
            </div>
          </div>
          <span
            class="navigationButton navigationButtonPrev navigationButtonPrev-0 material-symbols-outlined"
          >
            arrow_left_alt
          </span>
          <span
            class="navigationButton navigationButtonNext navigationButtonNext-0 material-symbols-outlined"
          >
            arrow_right_alt
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
